app/models/Signup.js

```diff
@@ -66,23 +67,28 @@ function parsePhoenixSignup(phoenixSignup) {
  */
 signupSchema.statics.lookupById = function (id) {
   const model = this;
-  const statName = 'phoenix: GET signups/{id}';
+  const notFoundMessage = 'No activity results for signup_id';
 
   return new Promise((resolve, reject) => {
     logger.debug(`Signup.lookupById:${id}`);
 
-    return phoenix.client.Signups.get(id)
-      .then((phoenixSignup) => {
-        stathat.postStat(`${statName} 200`);
-        logger.debug(`phoenix.Signups.get:${id} success`);
-        const data = parsePhoenixSignup(phoenixSignup);
+    return rogue.fetchActivityForSignupId(id)
+      .then((res) => {
+        // Because we get a success response with no results, throw an error to return a 404.
+        if (res.data.length < 1) {
+          throw new Error(notFoundMessage);
```

Comment:
This would be the perfect case to implement the NotFoundError in Conversations: https://github.com/DoSomething/gambit-conversations/blob/master/app/exceptions/NotFoundError.js.

Reply:
Once Conversations goes live -- the `POST /signups` endpoint that calls this `Signup.lookupById` function will be deprecated by our Conversations `POST /send-message` endpoint. I'd prefer to leave as is, as we'd just be getting rid of the newly added `NotFoundError` class and related endpoints in three weeks or so.